server/src/main/java/org/opensearch/common/time/OpenSearchDateTimeFormatter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.time;
+
+import java.text.Format;
+import java.text.ParsePosition;
+import java.time.ZoneId;
+import java.time.chrono.Chronology;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.time.temporal.TemporalQuery;
+import java.util.Locale;
+
+/**
+* Wrapper class for DateTimeFormatter{@link java.time.format.DateTimeFormatter}
+* to allow for custom implementations for datetime parsing/formatting
+ */
+class OpenSearchDateTimeFormatter implements OpenSearchDateTimePrinter {
+    private final DateTimeFormatter formatter;
+
+    public OpenSearchDateTimeFormatter(String pattern) {
+        this.formatter = DateTimeFormatter.ofPattern(pattern, Locale.ROOT);
+    }
+
+    public OpenSearchDateTimeFormatter(String pattern, Locale locale) {
+        this.formatter = DateTimeFormatter.ofPattern(pattern, locale);
+    }
+
+    public OpenSearchDateTimeFormatter(DateTimeFormatter formatter) {
+        this.formatter = formatter;
+    }
+
+    public OpenSearchDateTimeFormatter withLocale(Locale locale) {
+        return new OpenSearchDateTimeFormatter(getFormatter().withLocale(locale));
+    }
+
+    public OpenSearchDateTimeFormatter withZone(ZoneId zoneId) {
+        return new OpenSearchDateTimeFormatter(getFormatter().withZone(zoneId));
+    }
+
+    public OpenSearchDateTimeFormatter withChronology(Chronology chrono) {
```

Comment:
How is this used?

Reply:
Since, this is a wrapper over `DateTimeFormatter` I implemented all the necessary functions of `DateTimeFormatter` although right now it isn't being used. I removed it for brevity now. If required we can add it in wrapper in future